src/tests/ftest/launch.py

```diff
@@ -1199,28 +989,39 @@ def _pass_test(self, test_result, message=None):
             self.log.debug(message)
         test_result.status = TestResult.PASS
 
-    def _fail_test(self, test_result, fail_class, fail_reason, exc_info):
+    def _fail_test(self, test_result, fail_class, fail_reason, exc_info=None):
         """Set the test result as failed.
 
         Args:
             test_result (TestResult): the test result to mark as failed
             fail_class (str): failure category.
             fail_reason (str): failure description.
-            exc_info (OptExcInfo): return value from sys.exc_info().
+            exc_info (OptExcInfo, optional): return value from sys.exc_info(). Defaults to None.
         """
         self.log.error(fail_reason)
-        self.log.debug("Stacktrace", exc_info=True)
-
-        test_result.status = TestResult.ERROR
-        test_result.fail_class = fail_class
-        test_result.fail_reason = fail_reason
+        if exc_info is not None:
+            self.log.debug("Stacktrace", exc_info=True)
+
+        if test_result.fail_count == 0:
+            # Update the test result with the information abot the first error
+            test_result.status = TestResult.ERROR
+            test_result.fail_class = fail_class
+            test_result.fail_reason = fail_reason
+            if exc_info is not None:
+                try:
+                    # pylint: disable=import-outside-toplevel
+                    from avocado.utils.stacktrace import prepare_exc_info
+                    test_result.traceback = prepare_exc_info(exc_info)
+                except Exception:       # pylint: disable=broad-except
+                    pass
+        else:
+            # Additional errors only update the test result fail reason with a fail counter
+            plural = "s" if test_result.fail_count > 1 else ""
+            fail_reason = test_result.fail_reason.split(" (+")[0:1]
+            fail_reason.append(f"{test_result.fail_count} other failure{plural})")
+            test_result.fail_reason = " (+".join(fail_reason)
```

Comment:
FYI that if the fail reason ever contains `(+` this won't work as expected. E.g.
```
>>> 'apples (+1) oranges (+5 other failures)'.split(" (+")
['apples', '1) oranges', '5 other failures)']
>>> 'apples (+1) oranges (+5 other failures)'.split(" (+")[0:1]
['apples']
```

Reply:
That is understood.  We do control the `fail_reason` text so this shouldn't be an issue.  Since each fail reason is logged independently of the `fail_reason`, worse case the log will still contain all the error messages.